### DISCUSSION TITLE
Is there a way to prevent duplicate files among different records?

### DISCUSSION THREAD

**Discussion by amlcx** [pocketbase/pocketbase#2549] [Q&A] [answered]
Hi! I hope y'all are doing great.

I'm building an forum. However, In order to save disk space, I was thinking about getting the hash of files, and if it's been previously uploaded and exists in the filesystem, then skip the new upload and just point to the same file from different records.

Let's say the user tries to create a new record in the collection 'comments'; they attach picture.jpeg, from which we get the hash 'abc123'. Then, we can search for that hash in a separate collection (or, to simplify it even more, we can use ripgrep to traverse the filesystem or something); it it exists, then create the new comment and in the media field, point to the already existing file, otherwise, upload it normally.

Does pocketbase allow such behavior?
The other option I came up with was creating a microservice to handle all file upload operations, and remove all media fields from my pocketbase records, replacing them with a text file to store the URL.
Thanks in advance!

**Comment by ganigeorgiev** [comment, marked as answer]
> it it exists, then create the new comment and in the media field, point to the already existing file
> Does pocketbase allow such behavior?

There is no such functionality in PocketBase out of the box at the moment.
The general approach is similar to the one you've suggested - you can create a `files/media` collection that will hold the uploaded file and maybe a plain `text` field with its md5 hash. Then in the other collections you can create a `relation` field(s) pointing to the `files/media` collection.
The hash can be generated server-side by attaching to the [`OnRecord*Request` hooks](https://pocketbase.io/docs/event-hooks/#record-crud-api-hooks). Eventually in the near future there will be a generic file hooks (similar to the db model hooks) that will react on every file interaction done through filesystem abstraction, but that will be handled in https://github.com/pocketbase/pocketbase/issues/2387.

**Reply by amlcx** [reply]
I certainly appreciate your quick response. I'm glad to hear that, in the meantime I think the record hook option will be the way to go. Thanks again, PocketBase is such a wonderful project. I'll be checking the roadmap as well!
